app/packages/looker/src/elements/imavid/index.ts

```diff
@@ -276,14 +269,29 @@ export class ImaVidElement extends BaseElement<ImaVidState, HTMLImageElement> {
         }
 
         setTimeout(() => {
-          requestAnimationFrame(() =>
-            this.drawFrame(
-              Math.min(
-                frameNumberToDraw + 1,
-                this.framesController.totalFrameCount
-              )
-            )
-          );
+          requestAnimationFrame(() => {
+            const next = frameNumberToDraw + 1;
+
+            if (next > this.framesController.totalFrameCount) {
+              this.update(({ options: { loop } }) => {
+                if (loop) {
+                  this.drawFrame(1);
+                  return {
+                    playing: true,
+                    currentFrameNumber: 1,
```

Comment:
@sashankaryal could this be an arbitrary value? Is "first" frame key that should be used?